common/djangoapps/entitlements/models.py

```diff
@@ -0,0 +1,112 @@
+from django.db import models
+from django.contrib.auth.models import User
+from course_modes.models import CourseMode
+
+
+class CourseEntitlement(models.Model):
+    """
+    Represents a Student's Entitlement to a Course Run for a given Course.
+    """
+
+    user = models.ForeignKey(User)
+
+    parent_course_uuid = models.UUIDField(primary_key=True)
+
+    created = models.DateTimeField(auto_now_add=True, db_index=True)
+
+    updated = models.DateTimeField(auto_now=True)
+    # The date that an enrollment must be applied by
+    expiration = models.DateTimeField(null=False)
```

Comment:
Big question here for you and Deen.  How we expire entitlements may be more complicated than just a `datetime`.  We should talk to Deen and confirm this.

Reply:
I am thinking that we treat expiration like we do refunds.  It is a calculated policy rather than a stored field. Need product to confirm